realtime/redisPublisher.js

```diff
@@ -129,11 +129,14 @@ function publishSample(sampleInst, subjectModel, event, aspectModel) {
   const eventType = event || getSampleEventType(sampleInst);
   const useSampleStore =
     featureToggles.isFeatureEnabled('enableRedisSampleStore');
+
   return rtUtils.attachAspectSubject(sampleInst, useSampleStore, subjectModel,
     aspectModel)
   .then((sample) => {
-    publishObject(sample, eventType);
-    return sample;
+    if (sample) {
```

Comment:
This doesn't help shed any light on what circumstances would lead to publishSample getting called with an empty or invalid sampleInst argument. Please go deeper, go upstream, we need to understand how bad data is getting through this far.

Reply:
I did this https://github.com/salesforce/refocus/pull/599/commits/acfbd5dab37d4089dc70f851ca19881c0d36eb10 but @shriramshankar suggested these changes that's why I reverted back. @iamigo So what should we need now.